packages/grafana-ui/src/components/Table/FilterList.tsx

```diff
@@ -184,41 +185,41 @@ export const FilterList = ({
           <FilterInput placeholder="Filter values" onChange={setSearchFilter} value={searchFilter} />
         </Stack>
       )}
-      {!items.length && <Label>No values</Label>}
-      {items.length && (
-        <List
-          height={height}
-          itemCount={items.length}
-          itemSize={ITEM_HEIGHT}
-          width="100%"
-          className={styles.filterList}
-        >
-          {({ index, style }) => {
-            const option = items[index];
-            const { value, label } = option;
-            const isChecked = values.find((s) => s.value === value) !== undefined;
-
-            return (
-              <div className={styles.filterListRow} style={style} title={label}>
-                <Checkbox value={isChecked} label={label} onChange={onCheckedChanged(option)} />
-              </div>
-            );
-          }}
-        </List>
-      )}
-      {items.length && (
-        <Stack direction="column" gap={0.25}>
-          <div className={cx(styles.selectDivider)} />
-          <div className={cx(styles.filterListRow)}>
-            <Checkbox
-              value={selectCheckValue}
-              indeterminate={selectCheckIndeterminate}
-              label={selectCheckLabel}
-              description={selectCheckDescription}
-              onChange={onSelectChanged}
-            />
-          </div>
-        </Stack>
+      {!hasItems && <Label className={styles.noValuesLabel}>No values</Label>}
+      {hasItems && (
```

Comment:
As these two cases are exclusive, and the opposite of each other, i think it's better if it's a ternery.